.github/workflows/ci-lint.yml

```diff
@@ -17,28 +17,65 @@ on:
 
 env:
   GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  GOLANGCI_LINT_VERSION: v1.50
 
 jobs:
   golangci:
-    container: golangci/golangci-lint:latest
     name: Lint Go
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
         with:
           ref: ${{ inputs.ref }}
-      - name: Tidy
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ fromJson(inputs.version-set).go }}
+
+      - name: Lint pkg
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: ${{ env.GOLANGCI_LINT_VERSION }}
+          working-directory: pkg
+          args: --config ../.golangci.yml ./...
+      - name: Lint sdk
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: ${{ env.GOLANGCI_LINT_VERSION }}
+          working-directory: sdk
+          args: --config ../.golangci.yml ./...
+          # Only the first golangci-lint-action
+          # should set up the cache.
+          skip-cache: true
```

Comment:
👍 thorough.

nit: One way to read the comment here is that only `pkg` is cached, but I believe this action uses that each action will use and write to the same cached dir, and the post-job script will capture all three.

Reply:
Accurate. Let me cancel the merge and fix.